MobileManager/Services/AppResourceService.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using MobileManager.Controllers;
+using MobileManager.Database.Repositories.Interfaces;
+using MobileManager.Http.Clients.Interfaces;
+using MobileManager.Logging.Logger;
+using MobileManager.Models.App;
+using MobileManager.Services.Interfaces;
+
+namespace MobileManager.Services
+{
+    /// <summary>
+    /// AppResource service.
+    /// </summary>
+    public class AppResourceService : ControllerExtensions, IAppResourceService
+    {
+        private readonly IManagerLogger _logger;
+        private readonly IRestClient _restClient;
+        private readonly IRepository<AppResourceInfo> _appResourceRepository;
+        private readonly string _storagePath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:MobileManager.Services.AppResourceService"/> class.
+        /// </summary>
+        /// <param name="restClient">Rest client.</param>
+        /// <param name="appResourceRepository">appResourceRepository.</param>
+        /// <param name="logger">Logger.</param>
+        public AppResourceService(IRestClient restClient, IRepository<AppResourceInfo> appResourceRepository, IManagerLogger logger) : base(logger)
+        {
+            _logger = logger;
+            _restClient = restClient;
+            _appResourceRepository = appResourceRepository;
+            _storagePath = Path.Combine(Directory.GetCurrentDirectory(), "../_appFiles");
+        }
+
+        /// <inheritdoc />
+        public AppResourceInfo FindAppResource(string id)
+        {
+            return _appResourceRepository.Find(id);
+        }
+
+        /// <inheritdoc />
+        public bool DeleteAppResource(string id)
+        {
+            var appFolder = Path.Combine(_storagePath, id);
+            var result = _appResourceRepository.Remove(id);
+            Directory.Delete(appFolder, true);
+            return true;
+        }
+
+        /// <inheritdoc />
+        public async Task<AppResourceInfo> UploadAppResource(IFormFile file)
+        {
+            CreateStorageFolder();
+
+            var extension = Path.GetExtension(file.FileName).Trim('.').ToLower();
+            var hash = GetHashFromFile(file.OpenReadStream());
+            var folderPath = Path.Combine(_storagePath, hash);
+            var filePath = Path.Combine(folderPath, file.FileName);
+
+            if (!Directory.Exists(folderPath))

```

Comment:
this means that if the folder already exists, you just sent back the file info? shouldn't it return some error that you are uploading something that already exists?

Reply:
This is wanted behavior, this should be works like app storage, and if you want to upload file which is  already uploaded you just get information about already uploaded app instead of exception.